packages/tocco-ui/src/EditableValue/typeEditors/NumberEdit.js

```diff
@@ -6,53 +6,48 @@ import {parseLocalePlaceholder, convertStringToNumber} from '../utils'
 import {StyledEditableWrapper} from '../StyledEditableValue'
 import StyledNumberEdit from './StyledNumberEdit'
 
-export const limitValue = maxValueObject => values => {
-  const {formattedValue, floatValue} = values
-  return formattedValue === '' || floatValue <= maxValueObject
-}
+export const calculateMaxValue = (prePointDigits, postPointDigits, maxValue) => {
+  if (prePointDigits) {
+    const calculatedMaxValue = (10 ** prePointDigits) - (postPointDigits ? (1 / (10 ** postPointDigits)) : 1)
+    return maxValue && maxValue > calculatedMaxValue ? maxValue : calculatedMaxValue
+  }
 
-export const calculateMaxValue = (prePointDigits, postPointDigits) => {
-  const maxValueMinuend = (10 ** prePointDigits)
-  const maxValueSubtrahend = 1 / (10 ** postPointDigits)
-  return maxValueMinuend - maxValueSubtrahend
+  return maxValue
 }
 
-export const isAllowedIntegerValue = allowedIntegerObject => values => {
-  const {formattedValue, floatValue} = values
-  const {minValue, maxValue} = allowedIntegerObject
-  return formattedValue === '' || (floatValue >= minValue && floatValue <= maxValue)
+export const isAllowedValue = (prePointDigits, postPointDigits, minValue, maxValue) => ({floatValue}) => {
+  const calculatedMaxValue = calculateMaxValue(prePointDigits, postPointDigits, maxValue)
+
+  if (calculatedMaxValue && floatValue > calculatedMaxValue) {
+    return false
+  }
+
+  if (minValue && floatValue < minValue) {
+    return false
+  }
+
+  return true
 }
 
-export const returnAllowedIntegerObject = (minValue, maxValue) => {
-  return {minValue, maxValue}
+export const calculateDecimalScale = (postPointDigits, decimalScale) => {
```

Comment:
The more i think about it:
decimal scale and postpoint digits are the same, right? If yes, remove decimal scale from component and convert any "decimalScale" to "postPointDigits" in typeEditable file

Reply:
I don't think that is an improvement. As minValue/maxValue is more important there are inputs without postPointDigits. In these cases we control the number of decimals with decimalScale.